include/oneapi/dpl/pstl/utils.h

```diff
@@ -784,6 +785,35 @@ union __lazy_ctor_storage
     }
 };
 
+// To implement __min_nested_type_size, a general utility with an internal tuple
+// specialization, we need to forward declare our internal tuple first as tuple_impl.h
+// already includes this header.
+template <typename... T>
+struct tuple;
```

Comment:
I don't know the answer to this question, but does it make sense to put this inside of `tuple_impl.h` instead?

There are other tuple related helpers and utilities there.

Reply:
I had agreed with Dmitriy's comment in the original PR about moving it to the utils header: https://github.com/uxlfoundation/oneDPL/pull/1870#discussion_r1784951151 as it's not a core component of the tuple implementation and just a small utility.

I am open to reconsidering if you think it's best placed there.

Reply:
Aha, ok.  That's fine.  I forgot about that discussion.
We can leave it here.